internal/controller/model_pod.go

```diff
@@ -71,9 +81,111 @@ func (g *PodLoadBalancerModelGenerator) findPort(subset *corev1.EndpointSubset,
 	return -1, errPortNotFoundInSubset
 }
 
+func buildIngressRule(ingress *networkingv1.NetworkPolicyIngressRule) (rule model.IngressRule) {
+	rule.Ports = make([]model.PolicyPort, 0, len(ingress.Ports))
+	for _, port := range ingress.Ports {
+		newPort := model.PolicyPort{
+			Protocol: *port.Protocol,
+			EndPort:  port.EndPort,
+		}
+		if port.Port != nil {
+			newPort.Port = &port.Port.IntVal
```

Comment:
I added a test with multiple ports with port ranges in 9570ec8 and it succeeds.

I honestly don't quite understand yet how the issue in 158c8cd occurred, but a key difference is that in [ports.go:GetPorts()](https://github.com/cloudandheat/ch-k8s-lbaas/blob/158c8cdf6510d0d7b5423544084ee480ad84951a/internal/openstack/ports.go#L50), `fetched_ports` is filled with new ports repeatedly in a single run whereas is [model_pod.go:buildAllowedIngress()](https://github.com/cloudandheat/ch-k8s-lbaas/blob/1a18d9331db10b87cd7fcecd1b97f2c9d315d4e8/internal/controller/model_pod.go#L86), `&port.Port.IntVal` originates directly from the slice `allPolicies` in [model_pod.go:GenerateModel()](https://github.com/cloudandheat/ch-k8s-lbaas/blob/1a18d9331db10b87cd7fcecd1b97f2c9d315d4e8/internal/controller/model_pod.go#L160) that is filled only once per run.

My _guess_ is that the issue occurred because the underlying array of some slice was overwritten for the next page, but the pointers remained.

Do you think the unit test is sufficient? Do you think we should create a new pointer in any case, just to be sure?